bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java

```diff
@@ -1082,10 +1082,13 @@ public void setLength(long length) {
 
     private void checkFixedLength(long length) {
         if (arrayType != null && arrayType.getTag() == TypeTags.TUPLE_TAG) {
-            throw BLangExceptionHelper.getRuntimeException(BallerinaErrorReasons.INHERENT_TYPE_VIOLATION_ERROR,
-                    RuntimeErrors.CANNOT_CHANGE_TUPLE_SIZE);
-        }
-        if (((BArrayType) this.arrayType).getState() == ArrayState.CLOSED_SEALED) {
+            BTupleType tupleType = (BTupleType) this.arrayType;
+            if (tupleType.getRestType() == null
+                    || (tupleType.getRestType() != null && tupleType.getTupleTypes().size() > length)) {
```

Comment:
```suggestion
                    || tupleType.getTupleTypes().size() > length)) {
```
The not-null check is redundant since this would short circuit if the rest type was null.